test/integration/targets/nxos_ip_interface/tests/cli/sanity.yaml

```diff
@@ -0,0 +1,79 @@
+---
+- debug: msg="START TRANSPORT:CLI nxos_ip_interface sanity test"
+
+- set_fact: testint1="{{ nxos_int1 }}"
+- set_fact: testint2="{{ nxos_int2 }}"
+
+- name: "Put interface {{testint1}} into default state"
+  nxos_config: &intdefault1
+    lines:
+      - "default interface {{ testint1 }}"
+    provider: "{{ cli }}"
+  ignore_errors: yes
+
+- name: "Put interface {{testint2}} into default state"
+  nxos_config: &intdefault2
+    lines:
+      - "default interface {{ testint2 }}"
+    provider: "{{ cli }}"
+  ignore_errors: yes
+
+- name: "Make {{testint1}} a layer3 interface"
+  nxos_interface: &l3int1
+    interface: "{{ testint1 }}"
+    mode: layer3
+    description: 'Configured by Ansible - Layer3'
+    admin_state: 'up'
+    state: present
+    provider: "{{ cli }}"
+
+- name: "Make {{testint2}} a layer3 interface"
+  nxos_interface: &l3int2
+    interface: "{{ testint2 }}"
+    mode: layer3
+    description: 'Configured by Ansible - Layer3'
+    admin_state: 'up'
+    state: present
+    provider: "{{ cli }}"
+
+- name: Ensure ipv4 address is configured
+  nxos_ip_interface: &ipv4
+    interface: "{{ testint1 }}"
+    version: v4
+    state: present
+    addr: 20.20.20.20
+    mask: 24
+    provider: "{{ cli }}"
+  register: result
+
+- assert: &true
+    that:
+      - "result.changed == true"
```

Comment:
Right, I get that. I'm not expecting you to give me tests in a PR that are broken on your systems. What I am saying is that these are broken on our test platform, which is the only thing we have to test against, and which, even when we do get the 9k VM running, will still be used as a test image because (as far as I understand) there is literally nothing else for us to test with, and having the 7k gives us better coverage than not.

I don't think I'm being unreasonable when I ask that the tests that we get pass on the platforms we can test on. I understand that this means the tests might have some odd workarounds in, (as long as it does not negatively impact the tests' performance on real hardware, of course). What I do not want is a situation where we look at all our tests, and nxos is failing, and the default response is "Oh, that's alright, nxos is _always_ failing" and then we end up in this whole mess again.

With that out of the way, all I am looking for is:
* a setup task that sets the address to something other than 20.20.20.20 so that there is actually a change to detect
* some handling of ipv6 not trying to delete itself before it exists with tag `0` in the module. That probably means introspecting `existing` around line 396 to see if delta is in there. That's how I brute-forced my way out of it, but that may not be the best way.